go/oasis-test-runner/cmd/root.go

```diff
@@ -177,61 +317,103 @@ func runRoot(cmd *cobra.Command, args []string) error {
 	parallelJobCount := viper.GetInt(cfgParallelJobCount)
 	parallelJobIndex := viper.GetInt(cfgParallelJobIndex)
 
-	for index, v := range toRun {
-		n := v.Name()
-
-		if index%parallelJobCount != parallelJobIndex {
-			logger.Info("skipping test case (assigned to different parallel job)",
-				"test", n,
-			)
-			continue
-		}
-
-		if excludeMap[strings.ToLower(n)] {
-			logger.Info("skipping test case (excluded by environment)",
-				"test", n,
-			)
-			continue
-		}
+	// Parse test parameters passed by CLI.
+	var toRunExploded map[string][]scenario.Scenario
+	toRunExploded, err = parseTestParams(toRun)
+	if err != nil {
+		return fmt.Errorf("root: failed to parse test params: %w", err)
+	}
 
-		logger.Info("running test case",
-			"test", n,
-		)
-
-		childEnv, err := rootEnv.NewChild(n)
-		if err != nil {
-			logger.Error("failed to setup child environment",
-				"err", err,
-				"test", n,
-			)
-			return errors.Wrap(err, "root: failed to setup child environment")
-		}
+	// Run all test instances.
+	index := 0
+	for run := 0; run < numRuns; run++ {
+		for name, sc := range toRunExploded {
+			for i, v := range sc {
+				// Maintain unique scenario datadir.
+				runID := run*len(sc) + i
+				n := fmt.Sprintf("%s/%d", name, runID)
```

Comment:
Can we omit this in case we are only using a single run to avoid an extra path component?